v1/v1util/also.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1util
+
+import (
+	"io"
+)
+
+// ReadAndCloser implements io.ReadCloser by reading from a particular io.Reader
+// and then calling the provided "Close()" method.
+type ReadAndCloser struct {
```

Comment:
This might feel better as a method that takes a Reader and close func, and returns a ReadCloser. That avoids having to export the kind of ugly type.

Reply:
I'm going to make these private to `v1util` for now, since the aim is to use them for things like zip utilities, not this lower-level thing.